pandas/core/groupby/groupby.py

```diff
@@ -2241,6 +2242,10 @@ def _get_cythonized_result(
         for idx, obj in enumerate(self._iterate_slices()):
             name = obj.name
             values = obj._values
+            if is_extension_array_dtype(values.dtype) and is_integer_dtype(
```

Comment:
I think there is a more generic way to handle this? @jbrockmendel

Reply:
Yeah, agree this seems like an overly specific check that probably isn't catching other bugs. I tried just casting any ExtensionArray but got bunches of test failures (I think they had to do with datetimes).

Reply:
hmm this definitely seems not-great.  i think to do this right we need some EA method that gives the ndarray to use when we get here, and maybe a way to round-trip the results if the method is dtype-preserving.

Am I right in thinking that dt64tz (and maybe period?) get cast to i8 somewhere around here?

Reply:
Looking at this again I think it might make sense to update the `pre_processor` function: https://github.com/pandas-dev/pandas/blob/master/pandas/core/groupby/groupby.py#L1857 . Does that look right to you @jbrockmendel ?

Reply:
that seems plausible.  im hoping there's a way to do this with values_for_argsort that can be shared by EAs, and by other order-based methods

Reply:
Would that handle NAs in a way that would still give the right quantiles?